src/cider/tests/CiderNewAggHashTableTest.cpp

```diff
@@ -260,6 +261,236 @@ TEST_F(CiderNewAggHashTableTest, aggInt16Test) {
   CHECK_EQ(reinterpret_cast<int16_t*>(value2_check_ptr + 14)[0], 20);
 }
 
+TEST_F(CiderNewAggHashTableTest, aggUInt32Test) {
+  // SQL:
+  // SELECT SUM(int32), COUNT(int32), MIN(int32), MAX(int32) FROM table GROUP BY int32.
+  // The example below has 3 rows of data. Row number   key    value
+  //     0         1       10
+  //     1         2       20
+  //     2         1       30
+
+  // key of HT: int32
+  std::vector<SQLTypes> keys;
+  keys.push_back(SQLTypes::kINT);
+
+  // value of HT: SUM(int32)-int64 + COUNT(int32)-int32 + MIN(int32)-int32 +
+  // MAX(int32)-int32
+  uint32_t init_value_len = 20;
+  int8_t* init_value_ptr = allocator->allocate(init_value_len);
+  reinterpret_cast<int64_t*>(init_value_ptr)[0] = (int64_t)0;
+  reinterpret_cast<int32_t*>(init_value_ptr + 8)[0] = (int32_t)0;
+  reinterpret_cast<int32_t*>(init_value_ptr + 12)[0] =
+      std::numeric_limits<int32_t>::max();
+  reinterpret_cast<int32_t*>(init_value_ptr + 16)[0] =
+      std::numeric_limits<int32_t>::min();
+
+  cider::hashtable::AggregationHashTable aggregator(keys, init_value_ptr, init_value_len);
+
+  // Row0:
+  // Generate a key = 1
+  int8_t* key1_ptr = allocator->allocate(6);
+  reinterpret_cast<bool*>(key1_ptr)[0] = (bool)false;
+  reinterpret_cast<uint32_t*>(key1_ptr + 2)[0] = (uint32_t)1;
+  // Use get api and return value address
+  int8_t* value1_ptr = aggregator.get(key1_ptr);
+
+  // Check init value
+  CHECK_EQ(reinterpret_cast<int64_t*>(value1_ptr)[0], 0);
```

Comment:
`reinterpret_cast<int64_t*>(value1_ptr)[0]` -> `*reinterpret_cast<int64_t*>(value1_ptr)`
same for other

Reply:
and avoid us 0,8,12,,16, create a arg like offset_vector.